src/System.Windows.Forms.Design/tests/UnitTests/System/Windows/Forms/Design/ListBoxDesignerTests.cs

```diff
@@ -0,0 +1,153 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+namespace System.Windows.Forms.Design.Tests;
+
+public sealed class ListBoxDesignerTests
+{
+    [Fact]
+    public void IntegralHeight_WhenDockStyleNotFillLeftOrRight_ShouldSetShadowPropertyAndListBoxIntegralHeight()
+    {
+        using ListBoxDesigner designer = new();
+        using ListBox listBox = new();
+        designer.Initialize(listBox);
+
+        designer.IntegralHeight.Should().Be(true);
+
+        bool value = false;
+        designer.IntegralHeight = value;
+
+        designer.IntegralHeight.Should().Be(value);
+        listBox.IntegralHeight.Should().Be(value);
+    }
+
+    public static IEnumerable<object?[]> SetDockStyle_TestData()
+    {
+        yield return new object[] { DockStyle.Fill, false };
+        yield return new object[] { DockStyle.Left, false };
+        yield return new object[] { DockStyle.Right, false };
+        yield return new object[] { DockStyle.Bottom };
+        yield return new object[] { DockStyle.Top };
+    }
+
+    [Theory]
+    [MemberData(nameof(SetDockStyle_TestData))]
+    public void IntegralHeight_WhenDockStyleIsFillLeftOrRight_ShouldSetShadowPropertyAndNotListBoxIntegralHeight(DockStyle dockStyle, bool expected = true)
+    {
+        using ListBoxDesigner designer = new();
+        using ListBox listBox = new();
+        designer.Initialize(listBox);
+
+        designer.IntegralHeight.Should().Be(true);
+
+        listBox.Dock = dockStyle;
+        designer.IntegralHeight = expected;
+
+        designer.IntegralHeight.Should().Be(expected);
+        if (expected is false)
+        {
+            // When DockStyle is Fill/Left/Right, should set ShadowProperty and not set ListBox's IntegralHeight.
+            listBox.IntegralHeight.Should().NotBe(expected);
+        }
+        else
+        {
+            listBox.IntegralHeight.Should().Be(expected);
+        }
```

Comment:
I believe you can simplify to the following
```suggestion
        listBox.IntegralHeight.Should().Be(expected);
```